src/enum_clike.rs

```diff
@@ -35,8 +37,8 @@ impl LateLintPass for EnumClikeUnportableVariant {
                 if let Some(ref disr) = variant.disr_expr {
                     use rustc_const_eval::*;
                     let bad = match eval_const_expr_partial(cx.tcx, &**disr, EvalHint::ExprTypeChecked, None) {
-                        Ok(ConstVal::Integral(Usize(Us64(i)))) => i as u32 as u64 != i,
-                        Ok(ConstVal::Integral(Isize(Is64(i)))) => i as i32 as i64 != i,
+                        Ok(ConstVal::Integral(ConstInt::Usize(ConstUsize::Us64(i)))) => i as u32 as u64 != i,
+                        Ok(ConstVal::Integral(ConstInt::Isize(ConstIsize::Is64(i)))) => i as i32 as i64 != i,
```

Comment:
this change should not be necessary, but rather line `36` should be `use rustc_const_math::*`. I'm not sure why the unused import doesn't lint on that.